dotnetv3/S3/RestoreArchivedObjectExample/RestoreArchivedObjectExample/RestoreArchivedObject.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier:  Apache-2.0
+
+namespace RestoreArchivedObjectExample
+{
+    using System;
+    using System.Threading.Tasks;
+    using Amazon;
+    using Amazon.S3;
+    using Amazon.S3.Model;
+
+    /// <summary>
+    /// This example shows how to restore an archived object in an Amazon
+    /// Simple Storage Service (Amazon S3) bucket. The example was created
+    /// using the AWS SDK for .NET version 3.7 and .NET Core 5.0.
+    /// </summary>
+    public class RestoreArchivedObject
+    {
+        public static void Main()
+        {
+            string bucketName = "doc-example-bucket";
+            string objectKey = "archived-object.txt";
+
+            // Specify your bucket region (an example region is shown).
+            RegionEndpoint bucketRegion = RegionEndpoint.USWest2;
+
+            IAmazonS3 client = new AmazonS3Client(bucketRegion);
+            RestoreObjectAsync(client, bucketName, objectKey).Wait();
+        }
+
+        /// <summary>
+        /// This method restores a deleted object from an S3 bucket.
+        /// </summary>
+        /// <param name="client">The initialized Amazon S3 client used to call
+        /// RestoreObjectAsync.</param>
+        /// <param name="bucketName">A string representing the name of the
+        /// bucket where the object to restore was located.</param>
+        /// <param name="objectKey">A string representing the name of the
+        /// archived object to restore.</param>
+        public static async Task RestoreObjectAsync(IAmazonS3 client, string bucketName, string objectKey)
+        {
+            try
+            {
+                var restoreRequest = new RestoreObjectRequest
+                {
+                    BucketName = bucketName,
+                    Key = objectKey,
+                    Days = 2,
+                };
+                RestoreObjectResponse response = await client.RestoreObjectAsync(restoreRequest);
+
+                // Check the status of the restoration.
+                await CheckRestorationStatusAsync(client, bucketName, objectKey);
+            }
+            catch (AmazonS3Exception amazonS3Exception)
+            {
+                Console.WriteLine($"Error: {amazonS3Exception.Message}");
+            }
+        }
+
+        /// <summary>
+        /// This method retrieves the status of the object's restoration.
+        /// </summary>
+        /// <param name="client">The initialized Amazon S3 client used to call
+        /// GetObjectMetadataAsync.</param>
+        /// <param name="bucketName">A string representing the name of the
+        /// S3 bucket which contained the deleted object.</param>
+        /// <param name="objectKey">A string representing the name of the
+        /// archived object you want to restore.</param>
```

Comment:
archived or deleted

Reply:
This one is correct. It should be archived.